### PR TITLE
[Core] Propagate BindingContext changes to TitleView

### LINKED ISSUE #3596: Binding and GestureRecognizers not working inside TitleView on a NavigationPage
### Description
Binding elements of  TitleView on a NavigationPage not retrieving data from view model and also GestureRecognizers not fires by clicking on TitleView elements.

### Steps to Reproduce

1. Run the attached sample.
2. Browse to the items detail pages by selecting one item from list in main page.
3. Start editing on entry box with "Edit me" placeholder that not changes TitleView sub title.
4. Also by clicking TitleView inside toolbar nothing happens. 

### Expected Behavior
By editing text entry TitleView subtitle must be changed to editing text inside text entry.
By clicking on TitleView subtitle must be changed to "Command done".

### Actual Behavior
Nothing happens.

### Basic Information

- Version with issue: v3.2.0.637442-pre1 and  v3.2.0.729530-pre2
- Last known good version:
- IDE: VS2017 15.7.5
- Platform Target Frameworks: 
  - iOS:  
  - Android:  8.1
  - UWP:   

### Reproduction Link
[TitleViewBinding.zip](https://github.com/xamarin/Xamarin.Forms/files/2296115/TitleViewBinding.zip)

### LINKED ISSUE #3763: Gesture Commands do not fire in TitleView
### Description
I can use a `TapGestureRecognizer` on my `Image` within `TitleView`, but only with the `Tapped` event. When using `Command` it doesn't fire.

**Works:**
```
<NavigationPage.TitleView>
        <StackLayout Orientation="Horizontal" VerticalOptions="Center" Spacing="10">
            <Image Source="iconXamagon.png">
                <Image.GestureRecognizers>
                    <TapGestureRecognizer Tapped="Handle_Tapped"/>
                </Image.GestureRecognizers>
            </Image>
            <Label Text="3.2.0" FontSize="16" TextColor="Black" VerticalTextAlignment="Center" />
        </StackLayout>
        
    </NavigationPage.TitleView>
```

**Does NOT Work:**
```
<NavigationPage.TitleView>
        <StackLayout Orientation="Horizontal" VerticalOptions="Center" Spacing="10">
            <Image Source="iconXamagon.png">
                <Image.GestureRecognizers>
                    <TapGestureRecognizer  
                            Command="{Binding TapCommand, Mode=OneWay}"
                            CommandParameter="https://www.xamarin.com"/>
                </Image.GestureRecognizers>
            </Image>
            <Label Text="3.2.0" FontSize="16" TextColor="Black" VerticalTextAlignment="Center" />
        </StackLayout>
        
    </NavigationPage.TitleView>
```

The `TapCommand` above works elsewhere on the page. I would like it to work in this scenario also.

### Basic Information

- Version with issue: 3.2.0-pre3
- Tested iOS and Android

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###
If the BindingContext on the Page changes, the new BindingContext isn't propagated to the TitleView. These changes propagate that BindingContext to the TitleView in much the same way that it gets propagated to Menu Items

### Issues Resolved ### 

- fixes #3596 
- fixes #3763

### Platforms Affected ### 

- Core/XAML (all platforms)
- iOS
- Android
- UWP

### Testing Procedure ###
-Run Issue3763 in the control gallery and the instructions will indicate what you should see

### PR Checklist ###

- [x] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by PureWeen** [comment]
@samhouts  should we target this one for a 3.2 SR?

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3763.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+using Xamarin.UITest;
+using NUnit.Framework;
+using Xamarin.Forms.Core.UITests;
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 3763, "BindingContext on TitleView does not update after being initially set")]
+	public class Issue3763 : TestNavigationPage
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue3763.cs]: as these changes only target Core, a UnitTest would have been enough

File: Xamarin.Forms.Core/Page.cs [hunk 2]
```diff
@@ -421,5 +426,16 @@ bool ShouldLayoutChildren()
...
 			return _platformConfigurationRegistry.Value.On<T>();
 		}
+
+		internal void SetTitleView(View oldTitleView, View newTitleView)
+		{
+			if (oldTitleView != null)
+				oldTitleView.Parent = null;
+
+			if (newTitleView != null)
+				newTitleView.Parent = this;
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Page.cs]: does Parenting ensure that the InheritedBindingContext is set ? if so, 👍 , we're good

**Comment by AmrAlSayed0** [comment]
If you can rebase this to 3.2.0 SR1 please do so, 3.3.0 is too far away (6 weeks according to the wiki) to release a fix for an already released feature.

**Comment by StephaneDelcroix** [comment]
> @samhouts should we target this one for a 3.2 SR?

@PureWeen if the feature

File: Xamarin.Forms.Core/Page.cs (hunk 2 above)
Comment (PureWeen) [reply Xamarin.Forms.Core/Page.cs]: Setting the parent ensures that it gets set initially but it doesn't create a link that causes it to get updated in the future.  

The typical pattern around the code is for the Parent to know about the children and then propagate the BindingContext by overriding *BindingContextChanged*. That's how Label propagates to Span, VisualElements propagate to Gestures, and Spans propagate to gestures.

**Comment by samhouts** [comment]
> @samhouts should we target this one for a 3.2 SR?
Yes!

**Pull request merged**